src/node_crypto.cc

```diff
@@ -2853,205 +2829,6 @@ int SSLWrap<Base>::SetCACerts(SecureContext* sc) {
-                retry);
-    return 0;
-
-  } else if (BIO_should_read(bio)) {
-    DEBUG_PRINT("[%p] BIO: %s want read. should retry %d\n", ssl_, func, retry);
-    return 0;
-
-  } else {
-    char ssl_error_buf[512];
-    ERR_error_string_n(rv, ssl_error_buf, sizeof(ssl_error_buf));
-
-    HandleScope scope(ssl_env()->isolate());
-    Local<Value> exception =
-        Exception::Error(OneByteString(ssl_env()->isolate(), ssl_error_buf));
-    object()->Set(ssl_env()->error_string(), exception);
-
-    DEBUG_PRINT("[%p] BIO: %s failed: (%d) %s\n",
-                ssl_,
-                func,
-                rv,
-                ssl_error_buf);
-
-    return rv;
-  }
-
-  return 0;
-}
-
-
-int Connection::HandleSSLError(const char* func,
-                               int rv,
-                               ZeroStatus zs,
-                               SyscallStatus ss) {
-  ClearErrorOnReturn clear_error_on_return;
-
-  if (rv > 0)
-    return rv;
-  if (rv == 0 && zs == kZeroIsNotAnError)
-    return rv;
-
-  int err = SSL_get_error(ssl_, rv);
-
-  if (err == SSL_ERROR_NONE) {
-    return 0;
-
-  } else if (err == SSL_ERROR_WANT_WRITE) {
-    DEBUG_PRINT("[%p] SSL: %s want write\n", ssl_, func);
-    return 0;
-
-  } else if (err == SSL_ERROR_WANT_READ) {
-    DEBUG_PRINT("[%p] SSL: %s want read\n", ssl_, func);
-    return 0;
-
-  } else if (err == SSL_ERROR_WANT_X509_LOOKUP) {
-    DEBUG_PRINT("[%p] SSL: %s want x509 lookup\n", ssl_, func);
-    return 0;
-
-  } else if (err == SSL_ERROR_ZERO_RETURN) {
-    HandleScope scope(ssl_env()->isolate());
-
-    Local<Value> exception =
-        Exception::Error(ssl_env()->zero_return_string());
-    object()->Set(ssl_env()->error_string(), exception);
-    return rv;
-
-  } else if (err == SSL_ERROR_SYSCALL && ss == kIgnoreSyscall) {
-    return 0;
-
-  } else {
-    HandleScope scope(ssl_env()->isolate());
-    BUF_MEM* mem;
-    BIO *bio;
-
-    CHECK(err == SSL_ERROR_SSL || err == SSL_ERROR_SYSCALL);
-
-    // XXX We need to drain the error queue for this thread or else OpenSSL
-    // has the possibility of blocking connections? This problem is not well
-    // understood. And we should be somehow propagating these errors up
-    // into JavaScript. There is no test which demonstrates this problem.
-    // https://github.com/joyent/node/issues/1719
-    bio = BIO_new(BIO_s_mem());
-    if (bio != nullptr) {
-      ERR_print_errors(bio);
-      BIO_get_mem_ptr(bio, &mem);
-      Local<Value> exception = Exception::Error(
-          OneByteString(ssl_env()->isolate(),
-            mem->data,
-            mem->length));
-      object()->Set(ssl_env()->error_string(), exception);
-      BIO_free_all(bio);
-    }
-
-    return rv;
-  }
-
-  return 0;
-}
-
-
-void Connection::SetShutdownFlags() {
-  HandleScope scope(ssl_env()->isolate());
-
-  int flags = SSL_get_shutdown(ssl_);
-
-  if (flags & SSL_SENT_SHUTDOWN) {
-    Local<String> sent_shutdown_key = ssl_env()->sent_shutdown_string();
```

Comment:
You can remove `sent_shutdown_string`, `received_shutdown_string` and `onnewsessiondone_string` from env.h, this is the only place they're used.